### PR TITLE
[Compute] `az sig create/update`: Add new parameter `--soft-delete` to support soft delete

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Description**

Features to SIG Gallery and Gallery Image https://github.com/Azure/azure-cli/issues/18848

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -1019,12 +1019,17 @@ def load_arguments(self, _):
         c.argument('permissions', arg_type=get_enum_type(GallerySharingPermissionTypes), arg_group='Sharing Profile',
                    min_api='2020-09-30', is_experimental=True,
                    help='This property allows you to specify the permission of sharing gallery.')
+        c.argument('soft_delete', arg_type=get_three_state_flag(), min_api='2021-03-01', is_experimental=True,
+                   help='Enables soft-deletion for resources in this gallery, '
+                        'allowing them to be recovered within retention time.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1024]: Why is this new parameter `is_experimental` instead of `is_preview`?

Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1024]: Please use the first person voice in the help message
```suggestion
                   help='Enable soft-deletion for resources in this gallery, '
                        'allowing them to be recovered within retention time.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -1019,12 +1019,17 @@ def load_arguments(self, _):
...
     with self.argument_context('sig update') as c:
         c.ignore('gallery')
         c.argument('permissions', arg_type=get_enum_type(GallerySharingPermissionTypes), arg_group='Sharing Profile',
                    min_api='2020-09-30', is_experimental=True,
                    help='This property allows you to specify the permission of sharing gallery.')
-
+        c.argument('soft_delete', arg_type=get_three_state_flag(), min_api='2021-03-01', is_experimental=True,
+                   help='Enables soft-deletion for resources in this gallery, '
+                        'allowing them to be recovered within retention time.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:1032]: Same as above.

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 3]
```diff
@@ -3447,25 +3447,31 @@ def list_image_galleries(cmd, resource_group_name=None):
...
+                         location=None, no_wait=False, tags=None, permissions=None, soft_delete=None):
     Gallery = cmd.get_models('Gallery')
     location = location or _get_resource_group_location(cmd.cli_ctx, resource_group_name)
-
-    gallery = Gallery(description=description, location=location, tags=(tags or {}))
+    if soft_delete is not None:
+        policy = {'is_soft_delete_enabled': soft_delete}
+        gallery = Gallery(description=description, location=location, tags=(tags or {}), soft_delete_policy=policy)
+    else:
+        gallery = Gallery(description=description, location=location, tags=(tags or {}))
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:3474]: What happens when the value of `soft_delete` is `None` and passed into `Gallery`?

Comment (PARADISSEEKR) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:3474]: If the parameter passed in `soft_delete` is `None`, it is equivalent to `False`

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**